source/MaterialXCore/Node.cpp

```diff
@@ -252,14 +252,21 @@ vector<ElementPtr> NodeGraph::topologicalSort() const
 
     const vector<ElementPtr>& children = getChildren();
 
-    // Calculate in-degrees for all nodes, and enqueue those with degree 0.
-    std::unordered_map<ElementPtr, size_t> inDegree(children.size());
+    // Calculate in-degrees for all children.
+    std::unordered_map<ElementPtr, int> inDegree(children.size());
```

Comment:
Sure I can change that. The fix to handle negative values was just an extra precaution I added.. In theory it should never happen and would actually be an error in the graph (a node reached by downstream traversal should not have a zero in-degree). Would it be better to throw an exception if that happens?